pkg/systemd/overview-cards/cryptoPolicies.jsx

```diff
@@ -89,18 +101,34 @@ const CryptoPolicyDialog = ({
     setCurrentCryptoPolicy,
 }) => {
     const [error, setError] = useState();
+    const [inProgress, setInProgress] = useState(undefined);
     const [selected, setSelected] = useState(currentCryptoPolicy);
+    const [rebootRequired, setRebootRequired] = useState(currentCryptoPolicy);
 
     const setPolicy = (reboot) => {
+        const applyPolicy = reboot => {
+            if (reboot) {
+                cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: "require", err: "message" })
+                        .catch(error => setError(error));
+            } else {
+                cockpit.file(applyNeedRebootStamp, { superuser: "require" }).replace("\n")
+                        .then(() => close());
+            }
+        };
         cockpit.spawn(["update-crypto-policies", "--set", selected], { superuser: "require", err: "message" })
                 .then(() => {
                     setCurrentCryptoPolicy(selected);
-                    if (reboot) {
-                        cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: "require", err: "message" })
-                                .catch(error => setError(error));
+                    if (selected === "FIPS") {
+                        setRebootRequired(reboot);
+                        setInProgress(true);
+                        cockpit.spawn(["fips-mode-setup", "--enable"], { superuser: "require", err: "message" })
+                                .then(() => {
+                                    setInProgress(false);
+                                    applyPolicy(reboot);
+                                })
+                                .catch(error => { setInProgress(false); setError(error) });
```

Comment:
DRY -- Use `.finally()` for setInProgress(false), this then also simplifies the then/catch handlers to be functional instead of a code block.